### PR TITLE
Enable TCP keep alive while using HttpTransport

### PR DESCRIPTION
**Pull request by Deependra-Patel**
Reviewed here #687 
Extra diff is doing the same for Apache transport

### CODE REVIEW

File: util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java [hunk 1]
```diff
@@ -138,6 +145,9 @@ public static ApacheHttpTransport createApacheHttpTransport(
 
     HttpClientBuilder httpClientBuilder = ApacheHttpTransport.newDefaultHttpClientBuilder();
 
+    SocketConfig keepAliveSocketConf = SocketConfig.custom().setSoKeepAlive(true).build();
+    httpClientBuilder.setDefaultSocketConfig(keepAliveSocketConf);
```
Comment (mprashanthsagar) [inline util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java:149]: let's add a test to cover this path

**Comment by mprashanthsagar** [comment]
Please fix the presubmit build failures

**Comment by Deependra-Patel** [comment]
> Please fix the presubmit build failures

First time it failed due to maven dependency not able to download, on retry the next time it failed due to a transient test failing. It succeeds now.

File: util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java (hunk 1 above)
Comment (Deependra-Patel) [reply util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java:149]: I tried to find a way to test this but couldn't. So this config goes into building an [HttpClient](https://javadoc.io/doc/org.apache.httpcomponents/httpclient/latest/org/apache/http/client/HttpClient.html) which doesn't expose any method to expose this config.

I think it is fine to not unit test this config as it is obvious and we anyway don't have this in future versions. Also, for this version it is not default.

Comment (Deependra-Patel) [reply util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java:149]: wdyt?

File: util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java [hunk 2]
```diff
@@ -233,4 +246,68 @@ static URI parseProxyAddress(@Nullable String proxyAddress) {
...
+
+    public Socket createSocket(
+        InetAddress address, int port, InetAddress clientAddress, int clientPort)
+        throws IOException {
+      return setSocketKeepAlive(
+          wrappedSockedFactory.createSocket(address, port, clientAddress, clientPort));
+    }
+
+    private static Socket setSocketKeepAlive(Socket socket) throws SocketException {
+      socket.setKeepAlive(true);
```
Comment (medb) [inline util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java:309]: Please, add a release note for this in CHANGES.md

**Review by medb — CHANGES_REQUESTED** [review]

File: util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java (hunk 2 above)
Comment (Deependra-Patel) [reply util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java:309]: Done

File: util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java (hunk 1 above)
Comment (mprashanthsagar) [reply util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java:149]: ack

**Review by mprashanthsagar — APPROVED** [review]

**Pull request merged**